Lesson17/Ayse/contactBook.js

```diff
@@ -0,0 +1,331 @@
+  }else {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+
+let contacts5 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Anton', phone: '123-456-7899', email: 'anton@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+updateContact("Alice", "555-555-5555", "alice_new@gmail.com", contacts5);
+updateContact("Charlie", "111-111-1111", "charlie@gmail.com", contacts5);
+
+/*
+-----------------------------------------------------------
+  STEP 6: Remove a Contact
+-----------------------------------------------------------
+Function: removeContact(name)
+- Finds the index of the contact with 'name' using
+  findIndex() or a loop.
+- Splices it from the array if found.
+- Logs "Contact removed successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+console.log(' STEP 6: Remove a Contact');
+
+let contacts6 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+
+function removeContact(name, contactList) {
+  const contactIndex = contactList.findIndex(contact => contact.name === name);
+
+  if (contactIndex !== -1) {
+    contactList.splice(contactIndex, 1);
+    console.log(`Contact removed successfully.`);
+  }else {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+removeContact(`Alice`, contacts6);
+removeContact(`Charlie`, contacts6);
+
+/*
+-----------------------------------------------------------
+  STEP 7: Testing Our Functions
+-----------------------------------------------------------
+Below are some sample function calls to demonstrate the
+Contact Book in action.
+*/
+console.log('STEP 7: Testing Our Functions');
+
+let contacts7 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' },
+  { name: 'Charlie', phone: '222-555-8888', email: 'charlie@gmail.com' }
+];
+
+
+function addContact(name, phone, email, contactList) {
+  const existingContact = contactList.find(contact => contact.name === name);
+
+  if (existingContact) {
+    console.log(`${name} is already in the contact list.`);
+  } else {
+    contactList.push({ name, phone, email });
+    console.log(`${name} has been added to the contact list.`);
+  }
+  displayAllContacts(contactList);
+}
+
+addContact('Charlie', '333-333-3333', 'charlie@gmail.com', contacts7);
+///////////////////
+function viewContact(name, contactList) {
+  const contact = contactList.find(contact => contact.name === name);
+
+  if (contact) {
+    console.log(`Name: ${contact.name}, Phone: ${contact.phone}, Email: ${contact.email}`);
+  } else {
+    console.log(`No contact found with the name: ${name}`);
+  }
+  displayAllContacts(contactList);
+}
+
+viewContact('Alice', contacts7);
+///////////////////
+
+function updateContact(name, newPhone, newEmail, contactList) {
```

Comment:
Same for this and the functions below. Only keep on version of each function in this file.